core/src/main/java/com/linecorp/armeria/server/docs/DocService.java

```diff
@@ -297,13 +310,20 @@ private CompletableFuture<AggregatedHttpFile> loadVersions(Executor executor) {
             }, executor));
         }
 
-        private CompletableFuture<AggregatedHttpFile> loadSpecifications(Executor executor) {
-            return files.computeIfAbsent(specificationPath, key -> {
+        private CompletableFuture<ServiceSpecification> generateServiceSpecification() {
+            return CompletableFuture.supplyAsync(() -> {
+                final DocStringSupport docStringSupport = new DocStringSupport(services);
+                ServiceSpecification spec = generate(services);
+                spec = docStringSupport.addDocStrings(spec);
+                spec = exampleSupport.addExamples(spec);
+                return spec;
+            });
```

Comment:
Should we use the dedicated `executor` to generate `ServiceSpecification`?
```suggestion
            }, executor);
```

Reply:
Done :+1